src/transformers/modeling_utils.py

```diff
@@ -1777,10 +1788,29 @@ def save_pretrained(
 
         # Save the config
         if is_main_process:
-            model_to_save.config.save_pretrained(save_directory)
+            if not model_to_save._hf_peft_config_loaded:
```

Comment:
Since this has the potential to break code, are we 100% that the `model_to_save` will have the `._hf_peft_config_loaded` attribute? E.g. if we hit this line:

https://github.com/younesbelkada/transformers/blob/68a894a5875bfd958b8254afd3bbb23db9c2e813/src/transformers/modeling_utils.py#L2982

the attribute could not exist, right? Is it possible to hit that line?

Reply:
I was just asking because earlier we have `model_to_save = unwrap_model(self)`, which has the potential to return `self.module`. I don't know enough about transformers to tell if that's a potential issue, just wanted to be completely sure.

Reply:
maybe I can add `getattr(model, "_hf_peft_config_loaded", False)` to be on the safe zone

Reply:
Oh indeed. As far as I can tell that line is completely useless since a `PreTrainedModel` cannot be a `DistributedDataParallel` or a `DataParallel`.

Reply:
I dug more into it and this has basically always been here in one form or the other. But it is useless since no `PreTrainedModel` can have a `module` attribute, since once a model is wrapped for distributed training it changes class and stops inheriting from `PreTrainedModel`.